include/sys/device_mmio.h

```diff
@@ -16,6 +16,7 @@
 #define ZEPHYR_INCLUDE_SYS_DEVICE_MMIO_H
 
 #include <toolchain/common.h>
+#include <linker/sections.h>
```

Comment:
In #41555 I'm moving this `#include` before the previous one to fix a BUILD_ASSERT redefinition